.github/CONTRIBUTING.md

```diff
@@ -71,6 +71,38 @@ This project and everyone participating in it is governed by the
   smaller PRs that can be reviewed and merged independently. This makes it easier for
   reviewers to understand the changes and for maintainers to merge the PR.
 
+There are several code analysis tools, that are executing regularly: 
+
+- `checkstyle`
+- `PMD`
+- `spotbugs`
+- `openrewrite`
+
+#### Checkstyle
+Checkstyle enforces a consistent coding style in Kafka.
+You can run checkstyle using:
+
+Check: `mvn checkstyle:check`
+
+#### Rewrite
+The build system incorporates [Moderne](https://moderne.io/) rewrite capabilities
+for automated code transformations.
+
+- **Convention** (e.g., JUnit's naming rules)
+- **Refactor** safely (e.g., rename methods, migrate APIs)
+- **Modernize** (e.g., Java 8 → Java 17 features)
+- **Patterns** (e.g., replace `Vector` with `ArrayList`)
+
+Check: `mvn rewrite:dryRun`
+
+Apply: `mvn rewrite:run`
```

Comment:
You can print this in logs of job execution to hint new users on operating with failure

Reply:
please remove this update at .github/CONTRIBUTING.md

Reply:
yes of course, done.